### PR TITLE
PR: Add new font size for file switcher according to the OS

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

Added new font size for file switcher according to the OS in the fonts file in config.

This is the screenshot for Mac OS X
screen shot 2019-02-22 at 9 09 00 am

This is the screenshot for Linux
screen shot 2019-02-22 at 9 15 38 am

This is the screenshot for Windows

screen shot 2019-02-20 at 1 29 25 am

### Issue(s) Resolved

Part of #8580 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Juanis2112

### CODE REVIEW

**Comment by ccordoba12** [comment]
@juanis2112, please update the screenshots above to include the placeholder text in switcher search field. That makes font sizes easier to compare.

And also don't forget to include a Windows screenshot ;-)

File: spyder/widgets/fileswitcher.py [hunk 1]
```diff
@@ -19,6 +20,8 @@
 # Local imports
 from spyder.config.base import _
 from spyder.py3compat import iteritems, to_text_string
+from spyder.config.fonts import FILESWITCHER_FONT
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 please remove this line is not needed anymore and it is the cause of the failing tests :)

File: spyder/widgets/fileswitcher.py [hunk 2]
```diff
@@ -658,8 +660,20 @@ def setup_file_list(self, filter_text, current_path):
...
         # Build the text that will appear on the list widget
-        path_text_font_size = CONF.get('appearance', 'rich_font/size', 11)
-        filename_text_font_size = path_text_font_size + 2
+        rich_font = CONF.get('appearance', 'rich_font/size', 11)
+        if sys.platform == 'darwin':
+            path_text_font_size = rich_font
+            filename_text_font_size = path_text_font_size + 2
+        elif os.name == 'nt':
+            path_text_font_size = rich_font - 1
+            filename_text_font_size = path_text_font_size + 1
```
Comment (CAM-Gerlach) [inline spyder/widgets/fileswitcher.py]: Judging from the screenshots, Windows looks a point size or two smaller than the rest. Can you bump it up (by one?) to ensure it matches better?

Comment (juanis2112) [reply spyder/widgets/fileswitcher.py]: @ccordoba12 what do you think about this?

Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: I'm thinking to reduce Linux rich font size by one point, so I think Windows font size is fine.

Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: @CAM-Gerlach, you can adjust it to your own prefs, this is just the default.

Comment (CAM-Gerlach) [reply spyder/widgets/fileswitcher.py]: @ccordoba12 Yes, but its controlled by the global rich text size for *everything* in the UI, so it needs to be reasonably sized *relative* to the rest of the UI text. On Windows, it is, so if we bring Linux and macOS down one point as you state, then everything should be fine.

Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: I'm ok in doing that for Linux but I don't know about macOS.

@juanis2112, how things look for you in mac?

File: spyder/utils/icon_manager.py [hunk 3]
```diff
@@ -384,14 +384,16 @@ def get_icon(name, default=None, resample=False):
         return icon
 
 
-def icon(name, resample=False, icon_path=None):
+def icon(name, scalefactor=None, resample=False, icon_path=None):
```
Comment (ccordoba12) [inline spyder/utils/icon_manager.py]: Please change `scalefactor` to `scale_factor` because it's clearer.

File: spyder/utils/icon_manager.py [hunk 4]
```diff
@@ -402,24 +404,24 @@ def icon(name, resample=False, icon_path=None):
         return icon if icon is not None else QIcon()
 
 
-def get_icon_by_extension(fname):
+def get_icon_by_extension(fname, scalefactor):
```
Comment (ccordoba12) [inline spyder/utils/icon_manager.py]: `scalefactor` -> `scale_factor`

File: spyder/widgets/fileswitcher.py [hunk 5]
```diff
@@ -706,7 +706,15 @@ def setup_file_list(self, filter_text, current_path):
         for result in sorted(results):
             index = result[1]
             path = paths[index]
-            icon = ima.get_icon_by_extension(path)
+            if sys.platform == 'darwin':
+                scalefactor = 0.9
```
Comment (ccordoba12) [inline spyder/widgets/fileswitcher.py]: `scalefactor` -> `scale_factor`

File: spyder/plugins/explorer/widgets.py [hunk 6]
```diff
@@ -129,7 +129,7 @@ def icon(self, icontype_or_qfileinfo):
         else:
             qfileinfo = icontype_or_qfileinfo
             fname = osp.normpath(to_text_string(qfileinfo.absoluteFilePath()))
-            return ima.get_icon_by_extension(fname)
+            return ima.get_icon_by_extension(fname, 1.0)
```
Comment (ccordoba12) [inline spyder/plugins/explorer/widgets.py]: Please add here `scale_factor` to the keyword args, like this:

    return ima.get_icon_by_extension(fname, scale_factor=1.0)

File: spyder/widgets/fileswitcher.py (hunk 2 above)
Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: @juanis2112, to change the size of the rich font on Linux, please go to this line

https://github.com/spyder-ide/spyder/blob/master/spyder/config/fonts.py#L44

and change that value to `10`.

Then go to this line

https://github.com/spyder-ide/spyder/blob/master/spyder/config/main.py#L765

and change that value to `47.6.0`

Comment (juanis2112) [reply spyder/widgets/fileswitcher.py]: I think we could reduce size in macOS 1 point too.

Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: @juanis2112, you forgot to bump CONF_VERSION, as I described it here:

> Then go to this line
>
> https://github.com/spyder-ide/spyder/blob/master/spyder/config/main.py#L765
>
> and change that value to `47.6.0`

File: spyder/widgets/fileswitcher.py (hunk 5 above)
Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: @juanis2112, you forgot to address this comment.

File: spyder/widgets/fileswitcher.py (hunk 2 above)
Comment (juanis2112) [reply spyder/widgets/fileswitcher.py]: @ccordoba12 You are right! Sorry. I just did a new PR with that change.

File: spyder/widgets/fileswitcher.py (hunk 5 above)
Comment (juanis2112) [reply spyder/widgets/fileswitcher.py]: @ccordoba12 right! Done in the new PR.

Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: Thanks @juanis2112!

File: spyder/widgets/fileswitcher.py (hunk 2 above)
Comment (ccordoba12) [reply spyder/widgets/fileswitcher.py]: You are right! Sorry

Sure, no problem :+1:

> I just did a new PR with that change.

I think you meant a new commit ;-)

**Comment by ccordoba12** [comment]
@juanis2112, one last request: could you update the screenshots above with your latest changes? Thanks!

**Comment by juanis2112** [comment]
> @juanis2112, one last request: could you update the screenshots above with your latest changes? Thanks!

@ccordoba12  Done!

**Comment by ccordoba12** [comment]
Nice!! Thanks a lot @juanis2112!

**Pull request merged**